### ISSUE TITLE
Reopen on close

### ISSUE THREAD

**Issue by vampjaz** [websocket-client/websocket-client#95]
I am making an applcation using websocket.WebSocketApp, and i am working with an odd websocket server that doesn't seem to use ping/pong or any other method of keeping the socket open. I want to make a handler for on_close that reopens the socket, but i cant see a way to do that.

**Comment by nathancahill** [comment]
@liris I'm not seeing where run_forever() tries to reconnect to the server. Can you point me to the line?

**Comment by liris** [comment]
```python
while True:
    try:
       ws.run_forever()
    except:
        pass
```
ws.run_forerver re-connect at the second iteration.

**Comment by nathancahill** [comment]
Thanks!

**Comment by mtrojnar** [comment]
WRT https://github.com/websocket-client/websocket-client/issues/95#issuecomment-101850686:
1. `run_forever()` catches its exceptions, so `try:`/`except:` seems to be [dead code](https://en.wikipedia.org/wiki/Dead_code) here.
2. This code fails to gracefully handle the KeyboardInterrupt exception. https://github.com/websocket-client/websocket-client/commit/6531df5322d55474d11b4b08bedffa22c84f20a3 (available in websocket-client v0.54.0 and later) allows for simple:
``` python
while ws.run_forever():
    pass
```
Unfortunately, the [Debian python3-websocket package](https://packages.debian.org/sid/python3-websocket) is outdated...
The same issue is also discussed in #64 and #283.

**Comment by engn33r** [comment]
Closing this issue because it appears to have been resolved. If additional information is needed, please see issues #64, #283, or #580.
